tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -348,7 +348,7 @@ def update_leader(self):
         while retries > 0:
             for p in self.processors:
                 found = list(p.node.account.ssh_capture("grep \"Finished assignment for group\" %s" % p.LOG_FILE, allow_fail=True))
-                if len(found) == self.leader_counter[p] + 1:
+                if len(found) >= self.leader_counter[p] + 1:
```

Comment:
This is the actual fix. Everything else is just cleanup to address Intellij warnings.